extensions/wikia/VideoHandlers/tests/UpdateThumbnailTaskTest.php

```diff
@@ -0,0 +1,101 @@
+<?php
+/**
+ * Class UpdateThumbnailTaskTest
+ *
+ * @group MediaFeatures
+ */
+class UpdateThumbnailTaskTest extends WikiaBaseTest {
+
+	public function testThumbnailUploadOnNonExistingFile() {
+		$result = $this->callRetryThumbUpload(null);
+		$errors = $result->getErrorsArray();
+
+		$this->assertFalse($result->isGood());
+		$this->assertContains("File not found on wiki", $errors[0][0]);
+	}
+
+	public function testThumbnailUploadOnForeignFile() {
+		$file = $this->getVideoFileMock(false);
+
+		$result = $this->callRetryThumbUpload($file);
+		$errors = $result->getErrorsArray();
+
+		$this->assertFalse($result->isGood());
+		$this->assertContains("Cannot update foreign video thumbnail", $errors[0][0]);
+	}
+
+//	public function testThumbnailUploadSuccess() {
```

Comment:
Why is this code commented out? If it's wrong - we should fix it. If it's unnecessary - we should remove it. It'll stay in version control in case we need it back :-)